syne_tune/optimizer/schedulers/searchers/bayesopt/models/gp_model.py

```diff
@@ -433,22 +433,25 @@ def _draw_fantasy_values(self, state: TuningJobState) -> TuningJobState:
         )
 
     def configure_scheduler(self, scheduler):
-        from syne_tune.optimizer.schedulers.hyperband import HyperbandScheduler
+        from syne_tune.optimizer.schedulers.multi_fidelity import (
+            MultiFidelitySchedulerMixin,
+        )
 
         if isinstance(
             self._gpmodel, (IndependentGPPerResourceModel, HyperTuneJointGPModel)
         ):
-            assert isinstance(scheduler, HyperbandScheduler), (
+            assert isinstance(scheduler, MultiFidelitySchedulerMixin), (
                 "gpmodel of type IndependentGPPerResourceModel requires "
-                + "HyperbandScheduler scheduler"
+                "MultiFidelitySchedulerMixin scheduler"
             )
             # Likelihood of internal model still has to be created (depends on
-            # rung levels of scheduler). Note that ``max_t`` must be included
-            max_t = scheduler.max_t
-            if scheduler.rung_levels[-1] == max_t:
+            # rung levels of scheduler). Note that ``max_resource_level`` must be
+            # included
+            max_resource_level = scheduler.max_resource_level
+            if scheduler.rung_levels[-1] == max_resource_level:
                 rung_levels = scheduler.rung_levels
             else:
-                rung_levels = scheduler.rung_levels + [max_t]
+                rung_levels = scheduler.rung_levels + [max_resource_level]
```

Comment:
I see max_t and max_resource_level used. What's the difference and why do we need both?

Reply:
This is a bit of a legacy issue. `max_t` comes from Ray Tune code which was the original source for `HyperbandScheduler`. Since resource is quite different from time, I renamed this in later MF schedulers, but did not dare to change it for `HyperbandScheduler`, as people got used to it. In fact, it is almost always better to pass `max_resource_attr` to your scheduler ...

Reply:
To answer your question: `max_t` is a synonym for `max_resource_level`, which is used in `HyperbandScheduler`

Reply:
Would be great if we could get rid of max_t eventually (maybe 0.4)